src/map/pc_groups.cpp

```diff
@@ -3,462 +3,356 @@
-}
+using namespace rathena;
 
-/**
- * @retval NULL if not found
- * @private
- */
-static inline GroupSettings* name2group(const char* group_name)
-{
-	return (GroupSettings*)strdb_get(pc_groupname_db, group_name);
+const std::string PlayerGroupDatabase::getDefaultLocation() {
+	return std::string( conf_path ) + "/groups.yml";
 }
 
-/**
- * Loads group configuration from config file into memory.
- * @private
- */
-static void read_config(void)
-{
-	config_setting_t *groups = NULL;
-	const char *config_filename = "conf/groups.conf"; // FIXME hardcoded name
-	int group_count = 0;
-	
-	if (conf_read_file(&pc_group_config, config_filename))
-		return;
+uint64 PlayerGroupDatabase::parseBodyNode( const YAML::Node& node ){
+	uint32 groupId;
 
-	groups = config_lookup(&pc_group_config, "groups");
-
-	if (groups != NULL) {
-		GroupSettings *group_settings = NULL;
-		DBIterator *iter = NULL;
-		int i, loop = 0;
-
-		group_count = config_setting_length(groups);
-		for (i = 0; i < group_count; ++i) {
-			int id = 0, level = 0;
-			const char *groupname = NULL;
-			int log_commands = 0;
-			config_setting_t *group = config_setting_get_elem(groups, i);
-
-			if (!config_setting_lookup_int(group, "id", &id)) {
-				ShowConfigWarning(group, "pc_groups:read_config: \"groups\" list member #%d has undefined id, removing...", i);
-				config_setting_remove_elem(groups, i);
-				--i;
-				--group_count;
-				continue;
-			}
+	if( !this->asUInt32( node, "Id", groupId ) ){
+		return 0;
+	}
 
-			if (id2group(id) != NULL) {
-				ShowConfigWarning(group, "pc_groups:read_config: duplicate group id %d, removing...", i);
-				config_setting_remove_elem(groups, i);
-				--i;
-				--group_count;
-				continue;
-			}
+	std::shared_ptr<s_player_group> group = this->find( groupId );
+	bool exists = group != nullptr;
 
-			config_setting_lookup_int(group, "level", &level);
-			config_setting_lookup_bool(group, "log_commands", &log_commands);
-
-			if (!config_setting_lookup_string(group, "name", &groupname)) {
-				char temp[20];
-				config_setting_t *name = NULL;
-				snprintf(temp, sizeof(temp), "Group %d", id);
-				if ((name = config_setting_add(group, "name", CONFIG_TYPE_STRING)) == NULL ||
-				    !config_setting_set_string(name, temp)) {
-					ShowError("pc_groups:read_config: failed to set missing group name, id=%d, skipping... (%s:%d)\n",
-					          id, config_setting_source_file(group), config_setting_source_line(group));
-					continue;
-				}
-				config_setting_lookup_string(group, "name", &groupname); // Retrieve the pointer
+	if( !exists ){
+		if( !this->nodesExist( node, { "Name", "Level" })) {
+			return 0;
+		}
+
+		group = std::make_shared<s_player_group>();
+		group->id = groupId;
+		group->permissions = PC_PERM_NONE;
+	}
+
+	if( this->nodeExists( node, "Name" ) ){
+		std::string name;
+
+		if( !this->asString( node, "Name", name ) ){
+			return 0;
+		}
+
+		group->name = name;
+	}
+
+	if( this->nodeExists( node, "Level" ) ){
+		uint32 level;
```

Comment:
```suggestion
		uint16 level;
```
We limit GM level to 99, guess there's no reason to go this high.

Reply:
I will leave it as uint32 for the time being, as this is how it was before.
We can do another cleanup later.